src/object/srv_ec.c

```diff
@@ -125,36 +127,43 @@ ec_data_target(unsigned int dtgt_idx, unsigned int nr, daos_iod_t *iods,
 				ec_del_recx(iod, idx);
 				continue;
 			}
+			/* recx starts in this cell */
 			if (cell == dtgt_idx) {
-				uint32_t new_len = (cell + 1) *
-							oca->u.ec.e_len - so;
+				/* recx either extends beyond cell or
+				 * ends within cell
+				 */
+				uint64_t c_offset = recx_start % len;
+				uint64_t new_len = recx_size + c_offset >= len ?
+							len - c_offset :
```

Comment:
can we macrotize or inline function some of these types of calculations to avoid duplication?

Reply:
I'd like to leave it as it is for now.  The decision has been made to change to record-based EC, but I'm making these fixes in the byte-based and I'm just using power of 2 record sizes.